src/main/cpp/RLBotInterface/RLBotMessages/BoostUtilities/BoostUtilities.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef BOOSTUTILITIES_HPP
+#define BOOSTUTILITIES_HPP
+
+#include "../Messages.hpp"
+
+#include <boost\interprocess\ipc\message_queue.hpp>
+#include <boost/interprocess/shared_memory_object.hpp>
+#include <boost/interprocess/mapped_region.hpp>
+#include <boost\interprocess\sync\named_sharable_mutex.hpp>
+#include <boost\interprocess\sync\sharable_lock.hpp>
+
+#include "BoostConstants.hpp"
+
+// Adds the proto to the boost queue with message size checking and buffer size checking
+#define ADD_TO_BOOST_QUEUE(queue, proto, protosize, maxsize)	if (protosize >= maxsize) { \

```

Comment:
This is ok for now but I don't think it's a good long-term solution. Maybe it's better to put this code in a new function and call that function from wherever it needs to be called.

Reply:
I agree. See https://github.com/RLBot/RLBot/commit/70d733ea8344d6bb971533a5e14c3fa3fe1d75fc for an alternative technique.